docs/source/reference/api-server/api-server-tunning.rst

```diff
@@ -0,0 +1,152 @@
+
+        .. dropdown:: Why set the same value for the limits and requests?
+
+            We recommend setting the resources limits to the same value as the requests for the following reasons:
+            
+            * API server depends on the resources limit to detect the available resources and decide the maximum concurrency. Setting limits larger than the requests or omitting the limits will cause the API server make aggressive concurrency decisions and may cause high resource contention on the Kubernetes node.
+            * Setting the same value for the limits and requests ensures the QoS class of the API server pod being set to ``Guaranteed`` and reduce the chance of the pod being killed by the Kubernetes node when the node is under resource pressure.
+
+            In short, setting the same value for the limits and requests sacrifices the resources utilization for stability and predictability. Pivoting to other trade-off is also possible, but we recommend to keep the memory request and limit the same in production environment to avoid potential eviction caused by node memory pressure.
+
+    .. tab-item:: VM Deployment
+
+        For VM deployment, in-place vertically scaling the API server is not supported and the API server need to be terminated and recreated to apply the new resources. This means the current state of the API server will be lost. We recommend to create an new API server instance with the new resources and gradually migrate the workload to the new API server.
+
+        Refer to :ref:`sky-api-server-cloud-deploy` for how to deploy the new API server and modify the cluster configuration before running ``sky launch``:
+
+        .. code-block:: diff
+
+            resources:
+            -   cpus: 8+
+            -   memory: 16+
+            +   cpus: 16+
+            +   memory: 32+
+
+The following table shows the maximum concurrency for different resource configurations:
+
+.. list-table::
+   :widths: 1 1 3 2
+   :header-rows: 1
+
+   * - CPU
+     - Memory
+     - Max concurrency
+     - Suggested team size
+   * - 4
+     - 8Gi
+     - 8 Long requests, 11 Short requests
+     - 1~5 users
+   * - 16
+     - 32Gi
+     - 32 Long requests, 60 Short requests
+     - 1~15 users
+   * - 32
+     - 64Gi
+     - 64 Long requests, 145 Short requests
+     - 1~30 users
+   * - 64
+     - 128Gi
+     - 128 Long requests, 299 Short requests
+     - 10~50 users
+   * - 128
+     - 256Gi
+     - 256 Long requests, 589 Short requests
+     - 10~100 users
+
+Queuing requests and polling status asynchronously
+--------------------------------------------------
+
+There is no limit on the number of queued requests, i.e. despite increasing the allocated resources to improve the maximum concurrency, you can also submit requests with :ref:`async<async>` (``--async``) and poll the status asynchronously to avoid blocking. For example:
```

Comment:
```suggestion
There is no limit on the number of queued requests. To avoid request blocking, you can either (1) allocate more resources to increase the maximum concurrency (described above), or (2) :ref:`submit requests asynchronously <async>` (``--async``) and poll the status asynchronously.

For example:
```
Do we mean this?

I still find this revised section & the sec title confusing. What do we really want to say in this section? Lmk and I can try to rephrase.

It feels like "Use asynchronous requests as much as possible"?

Reply:
Yes, exactly. The background is that the request is synchronized by default to keep the behavior compatible with existing users from `0.8.0`. And we do want to recommend users to "Use asynchronous requests as much as possible", but I think it is worth mention the reason in the section title so how about "Use asynchronous requests to avoid blocking"?